sdk/go/common/apitype/environments.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apitype
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pulumi/esc"
```

Comment:
If we're taking a build time dependency on pulumi/esc is that now a (or soon going to be) a stable API where we'll endeavour not to make break changes?

Reply:
Yes. `esc`'s public API is intended to be stable.